src/Redis/Orleans.GrainDirectory.Redis/RedisGrainDirectory.cs

```diff
@@ -132,16 +135,16 @@ public async Task Initialize(CancellationToken ct = default)
             this.database = this.redis.GetDatabase();
 
             this.deleteScript = LuaScript.Prepare(
```

Comment:
note: I have ... let's say "regrets" that we allowed this API in; I almost never advocate for it - in most cases, the raw `ScriptEvaluate[Async]` API is preferable - in addition to being simpler, it can be more efficient *and* allows use of "native" redis Lua script, which makes it easier to test in isolation - i.e. you'd `KEYS[1]` or `ARGV[1]` etc; this also makes it necessary to think properly about what are keys and what aren't which is important because... I think this code gets it wrong

if we look at L97, we see `key = GetKey(address.GrainId)`, but: `GetKey` returns `string`, not `RedisKey`, which means: we might not be passing it via `KEYS`, but only by `ARGV`. This is critically important when using "cluster", which is the goal of this PR. It is probably sufficient (and necessary) to change the return type of `GetKey` to `RedisKey`, but if it was me... I just wouldn't use `LuaScript.Prepare` at all

Reply:
Does `ScriptEvaluateAsync` prepare the script first? My assumption is that re-parsing the Lua every time has a non-negligible perf cost

Reply:
@ReubenBond the library keeps a lookup of scripts it has seen; if it sees a new one, it uses `EVAL` for that iteration *but* it also issues `SCRIPT LOAD` in the background, and once it has seen a successful response from `SCRIPT LOAD`, *subsequent* uses of that same script are issued via `EVALSHA` - so: just like using `Prepare`, any overhead is amortized *but* it also doesn't need to do any reflection / emit which is how the prepared version is working with args etc.